agent/discover/discover.go

```diff
@@ -11,6 +11,8 @@ import (
 type Discovery func() (Discoverer, error)
 
 type Discoverer interface {
+	// Get Consul checkId string. Each discoverer has an unique id for that
+	GetId() string
```

Comment:
@dirkmueller it's a free text field, there is no format associated to it.

@arbulu89 we could comment as follows:
```suggestion
	// Returns an arbitrary unique string identifier of the discovery, so that we can associate it to a Consul check ID
	GetId() string
```